### ISSUE TITLE
Prolog language support

### ISSUE THREAD

**Issue by mcbloch** [hlissner/doom-emacs#4737]
**Describe the feature**
Support for the prolog language.

The swipl docs point to this website for emacs support. https://bruda.ca/emacs/prolog_mode_for_emacs

It would be nice to have this in doom as language option.

**System information**
<details><pre>
```
SYSTEM  info       ("Void Linux" . Linux 5.10.16_1 x86_64)
        shell      /bin/fish
        path       redacted
EMACS   dir        ~/.emacs.d/
        version    (27.1 Nov 29, 2020)
        buildopts  --with-x-toolkit=athena --without-toolkit-scroll-bars --without-dbus --without-gconf --without-gsettings --prefix=/usr --sysconfdir=/etc --sbindir=/usr/bin --bindir=/usr/bin --mandir=/usr/share/man --infodir=/usr/share/info --localstatedir=/var --with-file-notification=inotify --with-modules --with-jpeg --with-tiff --with-gif --with-png --with-xpm --with-rsvg --without-imagemagick --with-xml2 --with-gnutls --with-sound --with-m17n-flt --with-json --without-harfbuzz --without-cairo --with-libgmp --host=x86_64-unknown-linux-gnu --build=x86_64-unknown-linux-gnu 'CFLAGS=-fno-PIE -mtune=generic -O2 -pipe   -g' 'CPPFLAGS=   ' 'LDFLAGS=-no-pie -Wl,--as-needed    '
        features   XPM JPEG TIFF GIF PNG RSVG SOUND GLIB NOTIFY INOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB LUCID X11 XDBE XIM MODULES THREADS JSON PDUMPER LCMS2 GMP
        traits     (gui server-running envvar-file)
DOOM    dir        ~/.doom.d/
        version    (2.0.9 HEAD -> develop 5d9efe89b 2021-03-04 16:42:27 -0500)
        font       #<font-object -PfEd-DejaVu Sans Mono-normal-normal-normal-*-13-*-*-*-m-0-iso10646-1>
        theme      doom-one
        modules    (:input layout :completion company ivy :ui doom doom-dashboard doom-quit hl-todo indent-guides minimap modeline ophints (popup +defaults) unicode vc-gutter vi-tilde-fringe workspaces :editor (evil +everywhere) file-templates fold (format +onsave) lispy parinfer snippets :emacs dired electric undo vc :term vterm :checkers syntax :tools ansible debugger docker editorconfig (eval +overlay) lookup magit make terraform :lang cc clojure data elixir emacs-lisp gdscript (haskell +dante) json (java +meghanada) javascript latex ledger (markdown +grip) nim (org +roam) python (ruby +rails) rust sh web yaml :email mu4e :app irc :config (default +bindings))
        packages   ((evil-cleverparens) (aggressive-indent) (org-roam-server :recipe (:branch master)))
        unpin      (n/a)
        elpa       (n/a)
```

</pre></details>

**Comment by hlissner** [comment]
Hello! Thank you for the suggestion, but I'm afraid I must turn it down for a couple reasons:

1. Version 1.22 of prolog-mode is already included with Emacs. Only thing users need is:
   ```elisp
   (add-to-list 'auto-mode-alist '("\\.pl\\'" . prolog-mode))
   (add-to-list 'auto-mode-alist '("\\.m\\'" . mercury-mode))
   ```
2. Version 1.28 is available on that website, but is not hosted anywhere that our package manager can fetch it (not a public repo, not in ELPA/MELPA/marmalade, not on Emacsmirro or Emacsattic, etc). It cannot (and for security reasons, refuses to) fetch text files from arbitrary URLs.
3. A prospective module for prolog would amount to 2-4 lines of configuration, which is not enough to warrant its own module because users could trivially set it up themselves.

TL;DR Until prolog (1.28) is made available (e.g. a mirror) and requires more configuration to warrant a module, it doesn't make the cut. In any case, thanks for bringing it to my attention and sorry for the trouble!
